### PR TITLE
feat(design/test) : add storybook and configure tailwind support to it.

### LINKED ISSUE #22: Design/Test: Add Storybooks support to the app
This is a
[ ] bug
[x] feature request
[ ] documentation change
[ ] build/deploy enhancement request

Description:
Add support for storybooks (https://storybook.js.org/docs/react/get-started/introduction) 
To test it, add a story for the Header component

### PR DESCRIPTION
**Pull request by hussamkhatib**
Fixes #22 
moves the header from Home page to an individual component.(creates a components directory for it).
creates a sample story for header component.
The initial sample stories created by storybook are not removed yet.
storybookHeader

### CODE REVIEW

File: components/header/index.tsx
Comment (souvikbasu) [inline components/header/index.tsx:1]: creation of Header component is not part of this issue.
Let's merge the changes done for the Header component

File: postcss.config.js
Comment (souvikbasu) [inline postcss.config.js:4]: is this needed for storybooks? Can you please explain why these changes are made?

File: stories/Button.tsx
Comment (souvikbasu) [inline stories/Button.tsx:1]: Can we rename this component to ExampleButton so that we know that it was created only as an example implementation for storybooks?

File: styles/global.css
Comment (souvikbasu) [inline styles/global.css:1]: Can you please add this as a separate commit for tailwind integration with the app? This is independent from storybooks implementation

Comment (souvikbasu) [reply styles/global.css:1]: Ok, just saw that this is a separate commit. Resolving this comment

File: .storybook/main.js
Comment (souvikbasu) [inline .storybook/main.js:4]: These are not tailwind related changes as the commit message mentions. Can go as a separate commit

Comment (souvikbasu) [reply .storybook/main.js:4]: Similarly all other changes in this commit that is not tailwind related can go as part of another commit

**Review by souvikbasu — CHANGES_REQUESTED** [review]

File: components/header/index.tsx
Comment (hussamkhatib) [reply components/header/index.tsx:1]: I created it so that i can show the storybook sample as mentioned in the issue.
Will wait until @aryamantodkar pr is merged

File: postcss.config.js
Comment (hussamkhatib) [reply postcss.config.js:4]: No, i made this change while configuring taiwind to it .
This is the default configuration used by [Next.js](https://nextjs.org/docs/advanced-features/customizing-postcss-config).
i will remove this change .

File: stories/Button.tsx
Comment (hussamkhatib) [reply stories/Button.tsx:1]: All the stories inside `stories` directory are sample stories.
We can remove it as a whole ,if it's not required .
When running storybook locally , you can see they all come under Example dropdown. (see Image in pr for reference)
Do let me know what would be better to do.

File: .storybook/main.js
Comment (hussamkhatib) [reply .storybook/main.js:4]: okay , will look into it.

File: stories/Button.tsx
Comment (souvikbasu) [reply stories/Button.tsx:1]: I feel ExampleButton will still be better. Because in the IDE when somebody is trying to open the Button component of the app, both the components will show up and that might confuse the dev a little

**Pull request closed without merging**
